04week/algorithms.js

```diff
@@ -9,23 +9,136 @@ function getRandomInt(min, max) {
 }
 
 let arr = [];
+let flip = false;
 
-for (let i = 0; i < 1000; i++) {
-  arr.push(getRandomInt(0, 1000));
+for (let i = 0; i < 10; i++) {
+  arr.push(getRandomInt(0, 100));
 }
 
-function bubbleSort(arr) {
-  // Your code here
+function bubbleSort(ary) {
+  /*********************** WHITE BOARD NOTES **************************
+  // Using bubble sort alrgorithm to sort passed array.
+  // set flip flag equal to false
+  // Loop through array
+  //   Compare array[n] to array[n+1]
+  //   if array[n] is larger, then flip
+  //   set flip flag equal to true.
+  // End Loop
+  // if flip flag = true
+  //   set flip flag back to false
+  //   return bubbleSort(arr)
+  // else // array should be sorted
+  //   return arr
+  // end if
+  ********************************************************************/
+
+  for (let i = 0; i< ary.length; i++){
+    if (ary[i] > ary[i + 1]) {
+      flip = true;
+      // using the swap without temp algorithm!  Ex: ary[i] = 10, ary[i+1] = 6
+      ary[i+1] = ary[i+1] - ary[i];  //Ex. 6 - 10 = -4
+      ary[i]   = ary[i] + ary[i+1];  //Ex.  10 + (-4) = 6
+      ary[i+1] = ary[i] - ary[i+1];  //Ex.  6 - (-4) = 10
+    }
+  }
+  // flip? (flip = false, return bubbleSort(ary)) : return(ary); // doesn't work.
+  if (flip) {
+    flip = false;  // setting my check
+    return bubbleSort(ary);
+  } else {
+    return ary;
+  }
+}
+
+console.log('Bubble Sort');
+console.log(arr);
+console.log(bubbleSort(arr));
+
+function mergeSort(ary) {
+  /*********************** WHITE BOARD NOTES **************************
+  // Merge sort is a recursive algorithm.
+  // It is a divide in half approach.  Keep dviding until you get to
+  // only 1 element on either the left or right side.
+  // Now call merge.  Merge compares the two sub arrays and orders them.
+  // Ultimately, you get back to 2 sorted sub-arrays.  Merge will finally order them.
+  ********************************************************************/
+  if (ary.length < 2) return ary;
+
+  let middle = Math.floor(ary.length / 2);
+  let leftSide = ary.slice(0, middle);
+  let rightSide = ary.slice(middle);
+
+  return merge(mergeSort(leftSide), mergeSort(rightSide));
```

Comment:
Renee, I changed "let" vars to const where applicable.  Made changes in both mergeSort and binarySearch (not in bubbleSort as my commit msg indicates :))